velox/common/memory/SharedArbitrator.cpp

```diff
@@ -345,10 +358,25 @@ uint64_t SharedArbitrator::testingNumRequests() const {
   return numRequests_;
 }
 
+uint64_t SharedArbitrator::adjustedGrowCapacity(
+    const MemoryPool& pool,
+    uint64_t targetBytes) const {
+  uint64_t adjustedTargetBytes{0};
+  const auto capacity = pool.capacity();
+  if (capacity * 2 < 512ULL << 20) {
+    adjustedTargetBytes = capacity;
+  } else {
+    adjustedTargetBytes =
+        std::max(static_cast<uint64_t>(capacity / 4), 128ULL << 20);
+  }
+  return std::max(targetBytes, adjustedTargetBytes);
+}
+
 bool SharedArbitrator::growCapacity(
     MemoryPool* pool,
     const std::vector<std::shared_ptr<MemoryPool>>& candidatePools,
     uint64_t targetBytes) {
+  targetBytes = adjustedGrowCapacity(*pool, targetBytes);
```

Comment:
This might be the grow target under memory arbitration but we can quit if the actual required target bytes is satisfied before we reclaim from the others? Thanks!

Reply:
Yeah we can do that